R/list-simplify.R

```diff
@@ -0,0 +1,40 @@
+list_simplify <- function(x, simplify = NA, ptype = NULL) {
+  vec_assert(x, list())
+  if (length(simplify) > 1 || !is.logical(simplify)) {
+    abort("`simplify` must be `TRUE`, `FALSE`, or `NA`")
+  }
+  if (!is.null(ptype) && isFALSE(simplify)) {
+    abort("Must not specify `ptype` when `simplify = FALSE`")
+  }
+
+  if (isFALSE(simplify)) {
+    return(x)
+  }
+  strict <- !is.na(simplify)
+
+  # We choose not to simply data frames to preserve length invariants
```

Comment:
I think the size invariants will be preserved for data frames of size 1? I would expect `list_simplify()` to return a data frame of the same size as the input in that case.

Reply:
This comes from https://github.com/tidyverse/purrr/pull/782/files#diff-3c7e24ffc0f2dc10f8acf5c629ffa5da0a1479675f3c992d4afa586b4470bb67R515, which you wrote 😄  Maybe I misinterpreted your comment?

Reply:
hmm I think I was confused!

Reply:
Now, or then? 😆